pkg/store/bucket.go

```diff
@@ -1052,39 +1061,50 @@ func (b *bucketBlock) loadMeta(ctx context.Context, id ulid.ULID) error {
 
 func (b *bucketBlock) loadIndexCache(ctx context.Context) (err error) {
 	cachefn := filepath.Join(b.dir, block.IndexCacheFilename)
-
-	b.indexVersion, b.symbols, b.lvals, b.postings, err = block.ReadIndexCache(b.logger, cachefn)
-	if err == nil {
+	if err := b.loadIndexCacheFromFile(ctx, cachefn); err == nil {
 		return nil
 	}
 	if !os.IsNotExist(errors.Cause(err)) {
 		return errors.Wrap(err, "read index cache")
 	}
-	// No cache exists is on disk yet, build it from the downloaded index and retry.
+
+	// try to download index cache file from obj store.
```

Comment:
Comment as sentence.
```suggestion
	// Try to download index cache file from object store.
```